src/analyses/goto_check.cpp

```diff
@@ -900,23 +902,56 @@ void goto_checkt::pointer_overflow_check(
   if(!enable_pointer_overflow_check)
     return;
 
-  if(expr.id()==ID_plus ||
-     expr.id()==ID_minus)
+  if(expr.id() != ID_plus && expr.id() != ID_minus)
+    return;
+
+  if(expr.operands().size() > 2)
   {
-    if(expr.operands().size()==2)
+    // The overflow checks are binary!
+    // We break these up.
```

Comment:
`... e.g. for "a + b + c + d" we check "a + b", "(a + b) + c", "(a + b + c) + d"`